network/p2p/keyTranslator.go

```diff
@@ -13,6 +13,13 @@ import (
 	fcrypto "github.com/onflow/flow-go/crypto"
 )
 
+// This module is meant to help libp2p <-> flow public key conversions
+// Flow's Network Public Keys and LibP2P's public keys are a marshalling standard away from each other and inter-convertible.
+// Both are ECDSA public Keys on either the NIST P-256 curve or the "Bitcoin" secp256k1 curve, see https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#peer-ids
```

Comment:
1. The fact is libp2p fully supports secp256k1 keys, and Flow almost does. There are a couple of places that force the use of ECDSAP256 (the serde functions in `encodable/keys.go`, the one in `flow/identity.go), and one that enables it (the key generation in `cmd/bootstrap/utils/key_generation.go`). I can rephrase this comment, but this library would stay valid irrespective of the changes we'd enforce.

2. I value not confusing the user much more than respect to the historical sourcing, given that the mixup between the two curves is something we'll have to explain time and time again. I think the audience here is blockchain developers rather than academics, and hope the quotes around Bitcoin should be enough not to offend the later. Besides, strictly speaking SEC-2 is not only obscure, but non-canon, since SP 800-186 and FIPS 186-5 are *still* not final.

Reply:
I understand, that makes sense 👌